table.go

```diff
@@ -159,6 +165,15 @@ func (rt *RoutingTable) addPeer(p peer.ID, queryPeer bool) (bool, error) {
 		return false, ErrPeerRejectedHighLatency
 	}
 
+	// add it to the diversity filter for now.
+	// if we aren't able to find a place for the peer in the table,
+	// we will simply remove it from the Filter later.
```

Comment:
nit: This is a little awkward, especially since we can fix this by checking the diversity filter after we check if there's room instead of the other way around. This doesn't seem like it's necessarily a big deal though.